lib/router.ex

```diff
@@ -33,4 +39,22 @@ defmodule HTTPServer.Router do
     headers = Response.build_headers(body)
     Response.send_resp(status_code, body, headers)
   end
+
+  defp get_methods(handler) do
+    possible_methods = ["GET", "POST", "PUT", "PATCH", "DELETE"]
+
+    methods =
+      Enum.flat_map(possible_methods, fn method ->
+        case handler.handle(%Request{method: method}) do
+          {200, _body} -> [method]
+          {:error} -> []
+        end
```

Comment:
```suggestion
try do
  handler.handle(%Request{method: method})
  [method]
rescue
  FunctionClauseError -> []
end
```

@Aoxian What if this was a try rescue block and I was able to get rid of all the default error routes? The only reason I added those in there was because I read something in the documentation that suggested returning an `:error` was more "elixiry", but I was worried about the redundancy. Let me know if I'm missing a bigger picture here though.

Reply:
In functional programming, its always good to cover all possible matches.  If Elixir was a strongly typed functional language you could have a compile error if you do not have the default.  Is there a way to have this particular handle(req) method implemented for all Handlers by default?

Reply:
The try/catch would be a tradeoff, but it may be a justified one.  At least its a good first step. Perhaps there is an alternative design to use here to get the methods, like declaring them in the routes.ex file, then just parsing that file (not saying this is the correct answer, just pointing out there are alternatives).

Reply:
I've removed `get_methods` entirely and now declare the methods in the routes file! Let me know what you think about the change.